Monika After Story/game/script-topics.rpy

```diff
@@ -6937,6 +6937,307 @@ label monika_citizenship:
     m 2eka "But I want to make you happy, so...I'm going to do everything I can to keep bettering myself while I'm stuck here!"
     m 1eka "Thank you for listening to me vent, [player]."
     return
+    
+#Local variable to check if someone has skipped
+default persistent._mas_pm_monika_listened = None
+
+init 5 python:
+    addEvent(Event(persistent.event_database,eventlabel="monika_graduation",category=['school'],prompt="Graduation",random=True))
+
+label monika_graduation:
+    m 2ekc "You know, [player], one thing I really wish I got to experience is my high school graduation."
+    m "Pretty much my entire life so far has revolved around school."
+    m 4dkd "All those countless hours of studying, all the afterschool activities and clubs..."
+    m 4ekc "And in the end, after all that hard work, I never got to experience the fulfillment of actually graduating."
+    m 2dkd "Never got to walk across the stage and receive my diploma."
+    m "Never got to hear my name being announced and all my friends cheering."
+    m 2ekc "...It kinda feels like it was all for nothing."
+    m 2esd "I know all the things I learned along the way are what's really important."
+    m 2dkc "But it still feels like I missed out on something special."
+    m "..."
+    
+    #Went through and timed out on the menu twice
+    if persistent._mas_pm_monika_listened is False:
+        m 2lsc "Oh... Sorry, I hope I'm not boring you again..."
+        m 2esc "Let's forget about this and talk about something else, okay [player]?"
+        return "derandom"
+        
+    #Normal flow
+    else:
+        m 4eua "By the way, did you know I was the top student in my class?"
+        m 4rksdla "Ahaha... I don't mean to brag or anything, I only bring it up because as class valedictorian, I was supposed to give a speech at graduation."
+        m 2ekd "I spent so much time writing and practicing my speech, but no one ever got to hear it."
+        m 2eka "I was really proud of that speech, too."
+        m 2eua "I would love to recite it for you sometime, if you want to hear it~"
+        m 2eka "It's about a four minute speech, so just make sure you have enough time to listen to the whole thing."
+        m 4eua "Whenever you want to hear it, just let me know, okay?"
+        $ unlockEventLabel("monika_grad_speech_call")   
+        return
+
+init 5 python:
+
+    addEvent(Event(persistent.event_database,eventlabel="monika_grad_speech_call",category=['school'],prompt="Can I hear your speech now?",pool=True,unlocked=False,rules={"no unlock": None}))
```

Comment:
On its own, this the prompt for this is ambiguous. Can you change it to `Can I hear your graduation speech now?`